JASP-Desktop/backstage/opensavewidget.cpp

```diff
@@ -391,6 +376,42 @@ Utils::FileType OpenSaveWidget::getCurrentFileType()
 	return _currentFileType;
 }
 
+QString OpenSaveWidget::getCurrentFilePath()
+{
+	return _currentFilePath;
+}
+
+QString OpenSaveWidget::getDefaultOutFileName()
+{
+	QString path = getCurrentFilePath();
+	QString DefaultOutFileName="";
+	
+	if (path != "")
+	{
+		QString name =  QFileInfo(path).baseName();
+		QString ext = QFileInfo(path).suffix();
+		Utils::FileType filetype = Utils::getTypeFromFileName(path.toStdString());
+		switch (_mode)
+		{
+			case FileEvent::FileSave:
+				ext="jasp";
+				break;
+			case FileEvent::FileExportResults:
+				ext="html";
+				break;
+			case FileEvent::FileExportData:
+				if (ext=="") ext = "csv";
```

Comment:
Why here is there a 'if (ext=="")'. Probably because the file type can be .dat or .txt. But it should never be .html or .jasp. So I would be more explicit:
if (ext != 'dat' && ext != 'txt') ext = 'csv'